pkg/webhook/certs.go

```diff
@@ -151,15 +162,15 @@ func (cr *certRotator) refreshCerts(refreshCA bool, secret *corev1.Secret) (bool
 	}
 	writeFn := func() (bool, error) {
 		if err := cr.writeSecret(cert, key, caArtifacts, secret); err != nil {
-			return false, err
+			return false, nil
 		}
 		return true, nil
 	}
 	if err := wait.ExponentialBackoff(wait.Backoff{
-		Duration: 10 * time.Millisecond,
+		Duration: 30 * time.Millisecond,
 		Factor:   2,
 		Jitter:   1,
-		Steps:    10,
+		Steps:    5,
 	}, writeFn); err != nil {
 		log.Error(err, "max retries for writing of secret reached")
 		return false, err
```

Comment:
The reason for restarts is that the secret is not able to update here. It may restart once and succeed to write on the restart (in previous tests for the "manager" pod case). It may not be able to write and keep restarting (the "audit" pod case). The errors are the same: `“Operation cannot be fulfilled on secrets "gatekeeper-webhook-server-cert": the object has been modified; please apply your changes to the latest version and try again`